mne/io/base.py

```diff
@@ -1152,6 +1152,10 @@ def resample(self, sfreq, npad='auto', window='boxcar', stim_picks=None,
         if stim_picks is None:
             stim_picks = pick_types(self.info, meg=False, ref_meg=False,
                                     stim=True, exclude=[])
+        else:
+            stim_picks = _picks_to_idx(self.info, stim_picks, exclude=(),
+                                       with_ref_meg=False)
+
         stim_picks = np.asanyarray(stim_picks)
```

Comment:
We don't need this line anymore because in either part of the conditional above it's going to be an `ndarray` of int afterward